src/util/system.h

```diff
@@ -145,6 +145,8 @@ class ArgsManager
          * between mainnet and regtest/testnet won't cause problems due to these
          * parameters by accident. */
         NETWORK_ONLY = 0x200,
+        // This argument's value is sensitive (such as a password).
+        DONT_LOG = 0x400,
```

Comment:
I'd prefer to call this `SENSITIVE` or `PRIVATE` or similar to avoid a negated option (which might lead to double negatives), but more importantly make it agnostic of how the values are reported. If in the future someone adds a `getconfigargs` RPC, the internal wording "log" seems a bit off.